src/webapi/mutation_observer.rs

```diff
@@ -0,0 +1,259 @@
+    pub attribute_old_value: bool,
+
+    /// If `true` it will store the `CharacterData`'s old data in [`old_data`](enum.MutationRecord.html#variant.CharacterData).
+    pub character_data_old_value: bool,
+
+    /// If `Some` it will only observe the specified attributes. The attributes should be specified without a namespace.
+    ///
+    /// If `None` it will observe all attributes.
+    pub attribute_filter: Option< &'a [ &'a str ] >,
+}
+
+
+impl MutationObserver {
+    /// Returns a new `MutationObserver` with the given callback.
+    ///
+    /// The callback will be called with the following arguments when the observed DOM nodes change:
+    ///
+    /// 1. Vector of changes to the observed DOM nodes.
+    ///
+    /// 2. The `MutationObserver`.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver#Constructor)
+    pub fn new< F >( callback: F ) -> Self
+        where F: FnMut( Vec< MutationRecord >, Self ) + 'static {
+        js! (
+            return new MutationObserver( @{callback} );
+        ).try_into().unwrap()
+    }
+
+    /// Starts observing the `target`.
+    ///
+    /// The `MutationObserver`'s callback will be called with a list of
+    /// [`MutationRecord`](enum.MutationRecord.html) when the `target` changes.
+    ///
+    /// The `options` specifies which changes should be observed.
+    ///
+    /// Multiple different targets can be observed simultaneously (with the same or different `options`).
+    ///
+    /// If you call `observe` on the same `target` multiple times, it will replace the old `options`
+    /// with the new `options`. It will **not** call the callback multiple times for the same `target`.
+    ///
+    /// # Panics
+    ///
+    /// * At least one of
+    /// [`child_list`](struct.MutationObserverInit.html#structfield.child_list),
+    /// [`attributes`](struct.MutationObserverInit.html#structfield.attributes), or
+    /// [`character_data`](struct.MutationObserverInit.html#structfield.character_data) must be `true`.
+    ///
+    /// * If [`attribute_old_value`](struct.MutationObserverInit.html#structfield.attribute_old_value) is `true`, then
+    /// [`attributes`](struct.MutationObserverInit.html#structfield.attributes) must be `true`.
+    ///
+    /// * If [`attribute_filter`](struct.MutationObserverInit.html#structfield.attribute_filter) is `Some`, then
+    /// [`attributes`](struct.MutationObserverInit.html#structfield.attributes) must be `true`.
+    ///
+    /// * If [`character_data_old_value`](struct.MutationObserverInit.html#structfield.character_data_old_value) is `true`, then
+    /// [`character_data`](struct.MutationObserverInit.html#structfield.character_data) must be `true`.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver#observe())
+    pub fn observe< T: INode >( &self, target: &T, options: &MutationObserverInit ) {
```

Comment:
In this case you want to take `MutationObserverInit` instead of `&MutationObserverInit`.

As I've previously explained - yes, we usually want to take a borrow, *but* for a single-use helper structure like this which is used to emulate named parameters there isn't really a point.

Reply:
I'm still a bit fuzzy on when to borrow and when to own, and your comment isn't making that any clearer. 😛 But I'll change it.